helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/filter/range/FilterRangeConnectionSearcher.kt

```diff
@@ -0,0 +1,73 @@
+@file:Suppress("FunctionName")
+
+package com.algolia.instantsearch.helper.filter.range
+
+import com.algolia.instantsearch.core.connection.Connection
+import com.algolia.instantsearch.core.connection.ConnectionImpl
+import com.algolia.instantsearch.core.number.range.Range
+import com.algolia.instantsearch.helper.filter.range.internal.mapper
+import com.algolia.instantsearch.helper.searcher.SearcherSingleIndex
+import com.algolia.search.model.Attribute
+import com.algolia.search.model.response.ResponseSearch
+import com.algolia.search.model.search.FacetStats
+import com.algolia.search.model.search.Query
+import kotlin.reflect.KClass
+
+internal class FilterRangeConnectionSearcherImpl<T>(
+    private val filterRangeViewModel: FilterRangeViewModel<T>,
+    private val searcher: SearcherSingleIndex,
+    private val attribute: Attribute,
+    private val mapper: (Number) -> T
+) : ConnectionImpl() where T : Number, T : Comparable<T> {
+
+    val subscription: (ResponseSearch?) -> Unit = { response ->
+        filterRangeViewModel.computeBoundsFromFacetStats(attribute, response?.facetStats, mapper)
+    }
+
+    private fun <T> FilterRangeViewModel<T>.computeBoundsFromFacetStats(
+        attribute: Attribute,
+        facetStats: Map<Attribute, FacetStats>?,
+        mapper: (Number) -> T
+    ) where T : Number, T : Comparable<T> {
+
+        val facetStatsOfAttribute = facetStats?.get(attribute)
+        if (facetStatsOfAttribute == null) {
+            bounds.value = null
```

Comment:
wdyt if we put some logging here to guide the user and tell him something like: 
"couldn't find facetStats from response, make sure that the the attribute exists, or that the hits contain this facet" or something like this.

Reply:
I agree, however, this would require the introduction of a logging system, which doesn't not exist in the lib right now

Reply:
makes sense 👍